uitests/objectid/source/class/objectid/Application.js

```diff
@@ -53,6 +53,14 @@ qx.Class.define("objectid.Application", {
       this.getObject("exampleEditor/ctlrTitle").setModel(model);
 
       root.add(this.getObject("exampleEditor/container"), { left: 100, top: 250 });
+      
+      var button_panel = new qx.ui.toolbar.ToolBar();
+      button_panel.add(new qx.ui.toolbar.Button("Push Me"));
+      root.add(button_panel, { left: 100, top: 300 });
+      qx.core.Id.getInstance().register(button_panel, "buttons");
+      button_panel.setObjectId("buttons");
```

Comment:
Wouldn't it make sense to set the object id implicitly when registering the object? Specifying it twice seems redundant and might be a source of bugs. It would also save one line of code.

Reply:
the idea is that you can register shortcuts, so you could register A and C as top level, and then this code would be true: `qx.core.Id.getObject(“A/B/C/D/E”) === qx.core.Id.getObject(“C/D/E”)`
but when registering an object globally, you probably want to give it a different ID in that context, eg something more meaningful at a global level - so you have the option to override the ID for just globally, eg:
```
 qx.core.Id.register(C, “currentUser”);
/* .. snip .. */
qx.core.Id.getObject(“currentUser/edtName”).setValue(“George”);
qx.core.Assert.assertTrue (qx.core.Id.getObject("A/B/C") === qx.core.Id.getObject("currentUser"));
```
in other words, as the global objects set via `qx.core.Id.register()` are pointers into the ID heirarchy, the fact that you are registering something should not change the thing that you are registering

also, you could express your code as this:
```
       button_panel.setObjectId("buttons");
      qx.core.Id.getInstance().register(button_panel);
```
and `register()` will use the object’s objectId rather than assuming you need to override.